staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -213,8 +263,32 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, actual, err
 	}
 
-	if err := caseSensitiveJsonIterator.Unmarshal(data, obj); err != nil {
-		return nil, actual, err
+	// Handle strict mode.
+	if s.strict {
+		if yamlToJsonError != nil {
+			str := yamlToJsonError.Error()
+			if strings.Contains(str, `already set in map`) {
```

Comment:
not a fan of this, but i realized there is no other way to distinguish between duplicate vs unknown field errors.
let me know if you want me to remove the types and always return the generic NewStrictDecoderError instead.

bummer that popular libraries do not return typed errors.

Reply:
I'd write a comment about that we eventually want to do this in a better way, but can't atm as the libraries we use don't support structured error returns.